### PR TITLE
Readers/writers in same secure participant do not communicate

### PR DESCRIPTION
**Pull request by jrw972**

### CODE REVIEW

**Comment by mitza-oci** [comment]
Does this need the commit from #1723 or can it be kept separate?

File: dds/DCPS/DataReaderImpl.cpp [hunk 1]
```diff
@@ -259,13 +259,15 @@ DataReaderImpl::add_association(const RepoId& yourId,
...
-      if (this->qos_.durability.kind > DDS::VOLATILE_DURABILITY_QOS) {
-        info->waiting_for_end_historic_samples_ = true;
-      }
+    // Schedule timer if necessary
+    //   - only need to check reader qos - we know the writer must be >= reader
+    if (this->qos_.durability.kind > DDS::VOLATILE_DURABILITY_QOS) {
+      info->waiting_for_end_historic_samples_ = true;
+    }
+
+    remove_association_sweeper_->cancel_timer(writer_id);
```
Comment (mitza-oci) [inline dds/DCPS/DataReaderImpl.cpp:268]: Related changes for the "sweeper" are already in master from #1723. I suggest merging master into this branch when work on this one resumes.

**Comment by mitza-oci** [comment]
Test again after #2024 is merged

**Comment by jrw972** [comment]
Closed via #2381

**Pull request closed without merging**